### ISSUE TITLE
CA1822 (Mark members as static) codefix doesn't add static to partial definitions

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#5171]

**Comment by Siphonophora** [comment]
From looking at the linked PRs I think this won't surprise you, but I'm running into this in a .net 6 project.

``` csharp
public partial class Class1
{
    public partial int GetInt();
}

public partial class Class1
{
    public partial int GetInt() => new Random().Next(10); // CA1822
}
```

I can ctrl+. and use 'Make static' to convert to the code below, but that results in a CS0763 build error. 

```  csharp
public partial class Class1
{
    public partial int GetInt();
}

public partial class Class1
{
    public static partial int GetInt() => new Random().Next(10); // CS0763  ERROR
}
```

I'm not sure if #5172 should be in the analyzers i'm using, but it seems like there is still a false positive. Even if the code fix could normally update the partial definition, I am running into because a source generator is creating the partial definition. In that case, the code fix can't work and will just cause the build error. 

I think we wouldn't want CA1822 if the definition was in generated code. But maybe it makes sense to stop applying CA1822 to partial definitions as long as the code fix will create a build error.

**Comment by louis-z** [comment]
Just got a CA1822 false positive for a `partial` method implementation whose definition was auto-generated. The rule should ignore all `partial` methods, right?

**Comment by louis-z** [comment]
I had a look at the tests and wondered how `CA1822_PartialMethod_CanBeStaticAsync()`'s `fixedSource` could even compile... I extracted it to a source file, tried compiling it and got this error:

> CS0763: Both partial method declarations must be static or neither may be static

Now of course, in this particular case, we could probably make both the partial method definition and implementation static, but in real-life the definition might be generated by some tool which we have no control over. The partial method definition would then probably not be static, to keep implementation options open.

So @Youssef1313, shouldn't partial methods just always be ignored by the CA1822 rule?

**Comment by louis-z** [comment]
I just found out about https://github.com/dotnet/roslyn/pull/63447... Once the `IsGeneratedCode` property becomes available, I'm guessing this issue's resolution will come down to the following:
- If a partial method definition was generated, CA1822 will ignore the implementation.
- If a partial method definition was **not** generated, CA1822 will consider the implementation. If a diagnostic is reported, the codefix will adjust both the definition and implementation accordingly.

**Comment by Youssef1313** [comment]
@louis-z The tests in https://github.com/dotnet/roslyn-analyzers/pull/5172 doesn't seem to involve generated code.
